src/VirtualDom/Styled.elm

```diff
@@ -250,6 +257,139 @@ lazyHelp7 fn arg1 arg2 arg3 arg4 arg5 arg6 arg7 =
         |> toUnstyled
 
 
+type Scope
+    = Scope String
+
+
+encodeScope : Scope -> Json.Encode.Value
+encodeScope (Scope scope) =
+    Json.Encode.string scope
+
+
+uncurry : (a -> b -> c) -> ( a, b ) -> c
+uncurry f ( a, b ) =
+    f a b
+
+
+{-| Like map, but allows specifying an initial list to build on top of.
+-}
+mapOnto : (a -> b) -> List a -> List b -> List b
+mapOnto f xs init =
+    List.foldr (\x acc -> f x :: acc) init xs
+
+
+keyedLazy : (a -> ( String, Node msg )) -> a -> Node msg
+keyedLazy fn arg =
+    keyedLazyHelp fn arg
+        |> Unstyled
+
+
+keyedLazyHelp : (a -> ( String, Node msg )) -> a -> VirtualDom.Node msg
+keyedLazyHelp =
+    VirtualDom.lazy2
+        (\fn arg ->
+            fn arg
+                |> Tuple.mapFirst Scope
+                |> uncurry toScopedUnstyled
+        )
```

Comment:
🎨 I'd stylistically prefer this as a `let`, e.g.

```elm
\fn arg ->
    let
        ( key, node ) =
            fn arg
    in
    toScopedUnstyled (Scope key) node
```

Should run slightly faster too! 😄

Reply:
Done!